src/Node/MethodDeclaration.php

```diff
@@ -40,18 +40,38 @@ class MethodDeclaration extends Node implements FunctionLike {
         // FunctionBody
         'compoundStatementOrSemicolon'
     ];
+    
+    public function hasModifier(int $targetModifier) : bool {
+	    if ($this->modifiers === null) {
+		    return false;
+	    }
+	    foreach ($this->modifiers as $modifier) {
+		    if ($modifier->kind === $targetModifier) {
+			    return true;
+		    }
+	    }
+	    return false;
+    }
 
     public function isStatic() : bool {
-        if ($this->modifiers === null) {
-            return false;
-        }
-        foreach ($this->modifiers as $modifier) {
-            if ($modifier->kind === TokenKind::StaticKeyword) {
-                return true;
-            }
-        }
-        return false;
+        return $this->hasModifier(TokenKind::StaticKeyword);
     }
+	
+	public function isPublic() : bool {
+		return $this->hasModifier(TokenKind::PublicKeyword);
```

Comment:
`class X { function test() {}}` still has an implicitly public method `X::test()` - I can see this being misunderstood in applications using this helper, for that edge case.

This is low priority for me, though

Reply:
Yeah that's a good point. Could change `isPublic` to just check `!isPrivate` but it might be better to just expose `hasModifier`, I think the goal is to expose the AST but not necessarily "interpret" it...

Reply:
It might be useful to have a link to projects using tolerant-php-parser in the documentation, as examples for people writing their own applications/libraries.

E.g. there's https://github.com/tysonandre/tolerant-php-parser-to-php-ast (part of Phan), https://github.com/felixfbecker/php-language-server , etc.

The first one is notable in trying to convert all of the information provided by various node types.

Reply:
I made that change and just left `hasModifier`, FYI @inxilpro

Reply:
I could also see changing these to `hasPublicModifier()`/etc to be more clear about what question it's answering.